ui/src/masking.ts

```diff
@@ -8,16 +8,37 @@ import * as os from 'os';
 import { IActionContext, IParsedError } from "../index";
 import { parseError } from "./parseError";
 
+// No attempt will be made to mask usernames that are this length or less
+const UnmaskedUsernameMaxLength: number = 3;
+
 let _extValuesToMask: string[] | undefined;
 function getExtValuesToMask(): string[] {
     if (!_extValuesToMask) {
+        _extValuesToMask = [];
+    }
+    return _extValuesToMask;
+}
+
+let _usernameMask: RegExp | undefined | null = undefined;
+function getUsernameMask(): RegExp | undefined | null {
+    // _usernameMask starts out as undefined, and is set to null if building it fails or it is too short
+    // Specifically checking for undefined here ensures we will only run the code once
+    if (_usernameMask === undefined) {
         try {
-            _extValuesToMask = [os.userInfo().username];
+            const username = os.userInfo().username;
```

Comment:
Is there any good way of mocking this so we can write tests? I ran tests locally (using my own username, and also by overwriting at runtime with a shorter one) and everything worked but obviously that's only useful once.

Reply:
We could refactor these functions so that we pass in username as an argument. That way we can pass it in for testing. And add a wrapper function that we use at runtime that passes in `os.userInfo().username` for us.

Reply:
In order to preserve the compute-one-time-ness of it, we'd also need to expose `_usernameMask` in some way or allow it to be rest.

Reply:
Yeah, I think that sounds good. A function to set the value of `_usernameMask` to `undefined`. We can be clear about stating the function should be only used in tests.

Reply:
Added some tests with that approach.